### PR TITLE
[video] use action listener pattern for CPlayerController

### PR DESCRIPTION
**Pull request by xhaggi**
As the title said this refactors `CPlayerController` to use the action listener interface/pattern and register/unregister it `CApplication`. It also change `CPlayerController` to be Singleton so we can drop the member from `CApplication`.

@Paxxi mind taking a look

### CODE REVIEW

**Comment by Razzeee** [comment]
There's german in your commit message.

**Comment by xhaggi** [comment]
@Montellese thanks, rebased.
@Razzeee ups :grimacing: also changed.

File: xbmc/video/PlayerController.h [hunk 1]
```diff
@@ -46,6 +46,12 @@ class CPlayerController : public ISliderCallback
    */
   virtual void OnSliderChange(void *data, CGUISliderControl *slider);
 
+protected:
+  CPlayerController();
+  CPlayerController(const CPlayerController&);
+  CPlayerController& operator=(CPlayerController const&);
+  virtual ~CPlayerController();
```
Comment (Montellese) [inline xbmc/video/PlayerController.h]: The destructor shouldn't be protected.

Comment (xhaggi) [inline xbmc/video/PlayerController.h]: I've to disagree, if it's declared as a static member or static local within its own class, then the destructor may be private/protected.

File: xbmc/video/PlayerController.h [hunk 2]
```diff
@@ -29,11 +30,10 @@
  altering subtitles and audio tracks, changing aspect ratio, subtitle placement,
  and placement of the video on screen.
  */
-class CPlayerController : public ISliderCallback
+class CPlayerController : public ISliderCallback, public IActionListener
```
Comment (Memphiz) [inline xbmc/video/PlayerController.h]: Could you add a "virtual" to the OnAction method - for me this indicates that this is from the interface somehow.

File: xbmc/video/PlayerController.h (hunk 1 above)
Comment (Paxxi) [inline xbmc/video/PlayerController.h]: Yeah it's fine being private/protected

**Comment by Paxxi** [comment]
Add the virtual that @Memphiz mentioned and it's good to go imo

**Comment by xhaggi** [comment]
@Memphiz done

jenkins build this please

**Comment by xhaggi** [comment]
build erros are unrelated

**Pull request merged**